src/mlpack/tests/random_forest_test.cpp

```diff
@@ -607,3 +607,144 @@ TEST_CASE("ExtraTreesAccuracyTest", "[RandomForestTest]")
+  avg[1] = SequentialBootstrap<>::ComputeAverageUniqueness(
+      indM, 1, concurrency, invConcurrency);
+  avg[2] = SequentialBootstrap<>::ComputeAverageUniqueness(
+      indM, 2, concurrency, invConcurrency);
+
+  REQUIRE(avg(0) == Approx(5.0 / 6.0));
+  REQUIRE(avg(1) == Approx(0.75));
+  REQUIRE(avg(2) == Approx(1.0));
+}
+
+/**
+ * Test ComputeNextDrawProbabilities.
+ */
+TEST_CASE("ComputeNextDrawProbabilitiesTest", "[RandomForestTest]")
+{
+  arma::uvec phi1(2);
+  arma::umat indM(2 /* rows */, 3 /* cols */, arma::fill::zeros);
+
+  phi1[0] = 1u;
+
+  indM(0, 0) = 0;
+  indM(1, 0) = 2;
+  indM(0, 1) = 2;
+  indM(1, 1) = 3;
+  indM(0, 2) = 4;
+  indM(1, 2) = 5;
+
+  arma::vec concurrency(6, arma::fill::zeros);
+  concurrency[2] = 1;
+  concurrency[3] = 1;
+
+  arma::vec invConcurrency(concurrency.n_rows, arma::fill::ones);
+
+  // Compute the probabilities that observations 0, 1, 2 are drawn after
+  // observation 1 has already been drawn.
+  const arma::vec delta2(SequentialBootstrap<>::ComputeNextDrawProbabilities(
+      phi1, 1, concurrency, invConcurrency, indM));
+
+  REQUIRE(delta2(0) == Approx(5.0 / 14.0));
+  // Should have the lowest probability as it has already been drawn.
+  REQUIRE(delta2(1) == Approx(3.0 / 14.0));
+  // Should have the highest probability as this event does not overlap
+  // with others.
+  REQUIRE(delta2(2) == Approx(6.0 / 14.0));
+}
+
+/**
+ * Test ComputeSamples.
+ */
+TEST_CASE("ComputeSamplesTest", "[RandomForestTest]")
+{
+  arma::umat indM(2 /* rows */, 6 /* cols */, arma::fill::zeros);
+
+  indM(0, 0) = 0;
+  indM(1, 0) = 2;
+  indM(0, 1) = 2;
+  indM(1, 1) = 3;
+  indM(0, 2) = 4;
+  indM(1, 2) = 5;
+
+  indM(1, 3) = indM(1, 4) = indM(1, 5) = 5;
+
+  SequentialBootstrap bootstrap(indM);
+  const arma::uvec phi(bootstrap.ComputeSamples(6));
+
+  // Can only check that the next draw does not yield indices
+  // outside of the range of observations.
+  REQUIRE(phi(0) < 6);
+  REQUIRE(phi(1) < 6);
+  REQUIRE(phi(2) < 6);
+}
+
+/**
+ * Test SequentialBootstrap.
+ */
+TEST_CASE("SequentialBootstrapTest", "[RandomForestTest]")
+{
+  const arma::mat ds(10 /* rows */, 6 /* cols */, arma::fill::randu);
+  const arma::Row<size_t> labels{ 1, 0, 0, 0, 0, 0 };
+  const arma::vec weights(6, arma::fill::ones);
+  arma::umat indM(2 /* rows */, 6 /* cols */, arma::fill::zeros);
+
+  indM(0, 0) = 0;
+  indM(1, 0) = 2;
+  indM(0, 1) = 2;
+  indM(1, 1) = 3;
+  indM(0, 2) = 4;
+  indM(1, 2) = 5;
+
+  indM(1, 3) = indM(1, 4) = indM(1, 5) = 5;
+
+  SequentialBootstrap<> bootstrap(indM);
+  arma::mat bsDataset;
+  arma::Row<size_t> bsLabels;
+  arma::vec bsWeights;
+
+  bootstrap.Bootstrap<true>(
+      ds, labels, weights, bsDataset, bsLabels, bsWeights);
+
+  // Check that dimensions are the same.
+  REQUIRE(ds.n_rows == bsDataset.n_rows);
+  REQUIRE(ds.n_cols == bsDataset.n_cols);
+  REQUIRE(labels.n_cols == bsLabels.n_cols);
+  REQUIRE(weights.n_rows == bsWeights.n_rows);
+}
```

Comment:
Do you want to add one more test that actually trains a random forest using the sequential bootstrap?  With the tests that are already here, we know that the sequential bootstrap is behaving correctly (or at least not horribly wrong :smile:) but we should also check that we can create a `RandomForest` with a `SequentialBootstrap` object and that it gives decent enough results.

You could create a super simple synthetic dataset, that would be okay.  (Primarily I want to ensure that everything will compile when we use `SequentialBootstrap`.)